.github/workflows/build-test-e2e.yml

```diff
@@ -27,6 +27,16 @@ jobs:
     needs: [build]
     runs-on: 'macos-latest'
     timeout-minutes: 60
+    # Ensure that at most one playwright-tests job will run at a time in Sage-Bionetworks repo,
+    # but allow multiple playwright-test jobs to run concurrently in forked repos
+    # Per https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#concurrency
+    #  "When a concurrent job or workflow is queued, if another job or workflow
+    #   using the same concurrency group in the repository is in progress,
+    #   the queued job or workflow will be pending. Any previously pending job or
+    #   workflow in the concurrency group will be cancelled."
+    # Related discussion here: https://github.com/orgs/community/discussions/5435
+    concurrency:
+      group: ${{ github.repository_owner == 'Sage-Bionetworks' && format('${0}-${1}', github.workflow, '-playwright-tests') || format('${0}-${1}', github.run_id, matrix.shard ) }}
```

Comment:
GitHub's current concurrency implementation cancels previously queued pending jobs, rather than queuing and executing all pending jobs. So if more than two e2e workflow runs are triggered at the same time, depending on when each `playwright-tests` job starts, a subset or none of the e2e shards may run per workflow run. However, once all other jobs have finished running, the cancelled job(s) could be manually re-triggered, if needed.

For now, plan to monitor the e2e tests and the frequency of cancelled jobs, then decide whether another solution should be investigated.

Note: I applied concurrency at the `playwright-tests` job level, so that the `build` job (which builds SWC and runs unit tests on a Linux runner) will always run, regardless of how many e2e workflows have been triggered.